index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
+        numDisplay.textContent = rounder(Number(currentString));
+        firstNum = Number(currentString);
+        currentOp = "";
+        currentString = "";
+      }
+    } else {
+      return;
+    }
+  }
 
   for (let n = 0; n <= 9; n++) {
     numButtons[n].addEventListener("click", () => {
-      if (currentString.length <= 10) {
-        if (errLock == true) {
-          currentString = "";
-          errLock = false;
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        } else {
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        }
+      if (errLock == true) {
+        errLock = false;
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "" || currentString == "0") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString.match(/\d/g).length < 10) {
+        //If the currentString contains less than 10 digits (excluding decimal point and negative sign)
+        currentString = currentString + numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
       } else {
-        return;
+        currentString = currentString + numButtons[n].textContent;
+        clearCheck();
+        //Else, we will continue to add digits to the currentString, but NOT to the actual display (out of room)
       }
     });
   }
 
   numButtons[10].addEventListener("click", () => {
     if (errLock == true) {
-      return;
+      errLock = false;
+      currentString = "0" + numButtons[10].textContent;
+      numDisplay.textContent = currentString;
+      clearCheck();
     } else if (currentString.includes(".")) {
       return;
+    } else if (currentString == "") {
+      currentString = "0" + numButtons[10].textContent;
+      numDisplay.textContent = currentString;
+      clearCheck();
     } else {
       currentString = currentString + numButtons[10].textContent;
       numDisplay.textContent = currentString;
+      clearCheck();
     }
   });
 
   numButtons[11].addEventListener("click", () => {
-    if (errLock == true || currentString == "" || currentString == ".") {
+    if (errLock == true) {
       return;
-    } else if (currentString.length > 10) {
-      currentString = (Number(currentString) * -1).toExponential(5);
-      numDisplay.textContent = currentString;
+    } else if (currentString == "" && firstNum != "") {
+      currentString = (firstNum * -1).toString();
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
```

Comment:
Same comment here and below as earlier, about how you shouldn't use currentString to store the calculation result.

Also, since you're converting it back into a Number in both places you use it, you probably shouldn't convert the result to a string with `.toString()` in the first place.

Reply:
Updated. Created variable 'solution' to temporarily store result of operations instead of using 'currentString = currentString + '. Also eliminates superfluous .toString() method calls.